src/main/java/pingis/controllers/LiveChallengeController.java

```diff
@@ -0,0 +1,186 @@
+  }
+
+  @RequestMapping(value = "/newtaskpair")
+  public String newTaskPair(Model model) {
+    // TODO: add checks
+    logger.info("Displaying new task pair -form.");
+    return "newtaskpair";
+  }
+
+  @RequestMapping(value = "/createTaskPair", method = RequestMethod.POST)
+  public RedirectView createTaskPair(String testTaskName, String implementationTaskName,
+      String testTaskDesc, String implementationTaskDesc,
+      String testCodeStub, String implementationCodeStub,
+      long challengeId,
+      RedirectAttributes redirectAttributes) {
+    
+    Challenge currentChallenge = challengeService.findOne(challengeId);
+    Task testTask = gameplayService.generateTaskPairAndTaskInstance(testTaskName,
+        implementationTaskName,
+        testTaskDesc,
+        implementationTaskDesc,
+        testCodeStub,
+        implementationCodeStub,
+        currentChallenge);
+    redirectAttributes.addAttribute("taskId", testTask.getId());
+    redirectAttributes.addAttribute("testTaskInstanceId", 0L);
+    return new RedirectView("/playTurn/" + currentChallenge.getId());
+  }
+
+
+
+  @RequestMapping(value = "/playTurn/{challengeId}")
+  public RedirectView playTurn(Model model, @PathVariable Long challengeId,
+      RedirectAttributes redirectAttributes) {
+    
+    Challenge currentChallenge = challengeService.findOne(challengeId);
+    logger.info("Current Challenge fetched: " + currentChallenge);
+    
+    int index = gameplayService.getNumberOfTasks(currentChallenge) / 2;
+    logger.info("Highest index of tasks in current challenge: " + index);
+    
+    if (!gameplayService.isParticipating(currentChallenge)) {
+      logger.info("Not participating.");
+      
+      if (currentChallenge.getSecondPlayer() == null) {
+        currentChallenge.setSecondPlayer(userService.getCurrentUser());
+        challengeService.save(currentChallenge);
+        logger.info("Current user saved as a participant"
+            + " (second player) to current challenge.");
+      } else {
+        logger.info("Current user not a player in this challenge. Redirecting to /error.");
+        redirectAttributes.addFlashAttribute("message",
+            "this is not your challenge");
+        return new RedirectView("/error");
+      }
+    }
+    
+    TaskInstance unfinished = challengeService.getUnfinishedTaskInstance(currentChallenge);
+    logger.info("Unfinished task inside current challenge fetched: " + unfinished);
+
+    if (unfinished != null && unfinished.getUser().equals(userService.getCurrentUser())) {
+      logger.info("Found unfinished taskinstance owned by current user, redirecting to \"/task\"");
+      return new RedirectView("/task/" + unfinished.getId());
+
+    } else if (unfinished != null
+        && !unfinished.getUser().equals(userService.getCurrentUser())) {
+      logger.info("Unfinished taskinstance found, but not owned by the"
+          + " current user, redirecting to \"/user\"");
```

Comment:
No. There exists at most one unfinished task instance per challenge per user. Or at least should.